static/app/components/lazyLoad.tsx

```diff
@@ -39,84 +39,74 @@ type Props<C extends ComponentType> = Omit<React.ComponentProps<C>, 'route'> & {
-      setError(fetchError);
-    },
-    [setError]
-  );
 
-  const importComponent = props.component ?? props.route?.componentPromise;
+        return retryableImport(importComponent);
+      }),
+    [importComponent]
+  );
 
-  const fetchComponent = useCallback(async () => {
-    if (importComponent === undefined) {
-      return;
-    }
+  return (
+    <ErrorBoundary>
+      <Suspense
+        fallback={
+          <LoadingContainer>
+            <LoadingIndicator />
+          </LoadingContainer>
+        }
+      >
+        <LazyComponent {...(props as React.ComponentProps<C>)} />
+      </Suspense>
+    </ErrorBoundary>
+  );
+}
 
-    // If we're refetching due to a change to importComponent we want to make
-    // sure the current LazyComponent is cleared out.
-    setLazyComponent(null);
-
-    try {
-      const resolvedComponent = await retryableImport(importComponent);
-
-      // XXX: Because the resolvedComponent may be a functional component (a
-      // function) trying to pass it into the setLazyComponent will cause the
-      // useState to try and execute the function (because useState provides a
-      // "functional updates" invocation, see [0]) which is NOT what we want.
-      // So we use a functional update invocation to set the component.
-      //
-      // [0]: https://reactjs.org/docs/hooks-reference.html#functional-updates
-      setLazyComponent(() => resolvedComponent);
-    } catch (err) {
-      handleFetchError(err);
-    }
-  }, [importComponent, handleFetchError]);
-
-  // Fetch the component on mount and if the importComponent is updated
-  useEffect(() => void fetchComponent(), [fetchComponent]);
-
-  const fetchRetry = useCallback(() => {
-    setError(null);
-    fetchComponent();
-  }, [setError, fetchComponent]);
-
-  if (error) {
-    return (
-      <LoadingErrorContainer>
-        <LoadingError
-          onRetry={fetchRetry}
-          message={t('There was an error loading a component.')}
-        />
-      </LoadingErrorContainer>
-    );
+// Error boundaries currently have to be classes.
+class ErrorBoundary extends Component {
+  static getDerivedStateFromError(error: Error) {
+    return {
+      hasError: true,
+      error,
+    };
   }
 
-  if (!LazyComponent) {
-    return (
-      <LoadingContainer>
-        <LoadingIndicator />
-      </LoadingContainer>
-    );
-  }
+  state = {hasError: false, error: null};
 
-  if (LazyComponent === null) {
-    return null;
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    Sentry.withScope(scope => {
+      if (isWebpackChunkLoadingError(error)) {
+        scope.setFingerprint(['webpack', 'error loading chunk']);
+      }
+      scope.setExtra('errorInfo', errorInfo);
+      Sentry.captureException(error);
+    });
+
+    // eslint-disable-next-line no-console
+    console.error(error);
   }
 
-  return <LazyComponent {...(props as React.ComponentProps<C>)} />;
+  fetchRetry = () => this.setState({hasError: false});
```

Comment:
damn this is slightly less intuitive -- but I assume it's that re-rendering the component with Suspense will try and load it again right

Reply:
yeah exactly